shared/controlflow/codeql/controlflow/Guards.qll

```diff
@@ -1104,6 +1105,24 @@ module Make<
         )
       }
 
+      private predicate validReturnInCustomGuardToRank(
+        int rnk, NonOverridableMethod m, ParameterPosition ppos, GuardValue retval, GuardValue val
+      ) {
+        validReturnInCustomGuard(_, _, m, ppos, retval, val) and rnk = 0
```

Comment:
It's the `validReturnInCustomGuard(_, _, m, ppos, retval, val)` that ensures that this becomes a `forex` and not a `forall`, right? If so, that might deserve a comment.

Reply:
Yes. And the fact that we've pushed the constraint
```
          not exists(GuardValue notRetval |
            exprHasValue(ret, notRetval) and
            disjointValues(notRetval, retval)
          )
```
into it.

Reply:
I've added a commit with an elaborating comment.